### PR TITLE
Github action to create release in OctopusDeploy

### PR DESCRIPTION
**Pull request by rain-on**
Added a github action which, when a release is created, creates a release in the Octopus server for the built package.

The github action can also be run manually - however a link to the Github release must be supplied.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        if ["${{github.event_name}}" = "release"]; then
+           RELEASE_NOTES=curl ${{ github.event_path }} | jq '.body'
+           echo "::set-output name=release_notes::RELEASE_NOTES"
+        else
+           RELEASE_NOTES=curl ${{ inputs.release-url }} | jq '.body'
+           echo "::set-output name=release_notes::RELEASE_NOTES"
+        fi
+
+    - name: Create a release in Octopus Deploy 🐙
+      uses: OctopusDeploy/create-release-action@v1.0.2
```
Comment (long-sean) [inline .github/workflows/release.yml:60]: Hardcoded version number?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+# Controls when the action will run.
+on:
+  release:
+    types: [created]
+
+  workflow_dispatch:
+    inputs:
+      release-url:
+        description: "The URL of the github release being created in OD"
```
[the remark this answers is not in the document]
Comment (rain-on) [reply .github/workflows/release.yml:11]: done

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        id: create-package
+        run: |
+        ./gradlew distZip
+        PACKAGE_NAME=`./gradlew -q packageName`
+        echo "::set-output name=package-created::$PACKAGE_NAME"
+        version=`./gradlew properties | grep ^version | awk '{print $2;}'`
+        echo "::set-output name=version::$version"
+
+    - name: Push a package to Octopus Deploy 🐙
+      uses: OctopusDeploy/push-package-action@<version>
```
Comment (long-sean) [inline .github/workflows/release.yml:41]: Is "version" available here, I do not know how variables are scoped in GH actions

**Review by long-sean — COMMENTED** [review]
Left a few queries, some due to my lack of familiarity with GH actions

File: .github/workflows/release.yml (hunk 1 above)
Comment (rain-on) [reply .github/workflows/release.yml:60]: specifying a fixed version seems to be the normal way of doing things

File: .github/workflows/release.yml (hunk 3 above)
Comment (rain-on) [reply .github/workflows/release.yml:41]: yeah whoops - hard coded.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      - name: Build with gradle
+        run: ./gradlew build test
+
+      - name: Create Plugin Zip Deployment
+        id: create-package
+        run: |
+        ./gradlew distZip
+        PACKAGE_NAME=`./gradlew -q packageName`
+        echo "::set-output name=package-created::$PACKAGE_NAME"
+        version=`./gradlew properties | grep ^version | awk '{print $2;}'`
```
[the remark this answers is not in the document]
Comment (rain-on) [reply .github/workflows/release.yml:37]: I was going with:
* bash variables conform to bash casing (typically upper) 
* github action variables seem to use snake-case, all lower

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+          api_key: ${{ secrets.OCTOPUS_APIKEY }}
+          packages: ${{ step.create-package.package-created }}
+          server: ${{ secrets.OCTOPUS_SERVER }}
+
+      - name: Fetch Release Notes
+        id: fetch-release-nodes
+        run: |
+          if ["${{github.event_name}}" = "release"]; then
+             RELEASE_NOTES=curl ${{ github.event_path }} | jq '.body'
+             echo "::set-output name=release-notes::RELEASE_NOTES"
```
Comment (long-sean) [inline .github/workflows/release.yml:52]: missing `$` variable prefix

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+          else
+             RELEASE_NOTES=curl ${{ inputs.release-url }} | jq '.body'
+             echo "::set-output name=release-notes::RELEASE_NOTES"
```
Comment (long-sean) [inline .github/workflows/release.yml:55]: missing `$` variable prefix

**Review by long-sean — COMMENTED** [review]
Missing variable prefix on RELEASE_NOTES

**Review by long-sean — APPROVED** [review]

**Pull request merged**